iXBRLViewerPlugin/viewer/src/js/fact.js

```diff
@@ -251,3 +251,19 @@ Fact.prototype.footnotes = function () {
 Fact.prototype.isHidden = function () {
     return this._ixNode.wrapperNode.length == 0;
 }
+
+Fact.prototype.widerConcepts = function () {
+    if (!this._report.usesAnchoring()) {
+        return [];
+    }
+    var concepts = [];
+    const wn = this._report.data.rels["w-n"];
+    for (const elr in wn) {
+        for (const src in wn[elr]) {
+            if ($.map(wn[elr][src], (target) => target.t).includes(this.conceptName())) {
```

Comment:
I guess this is loop is O(anchors), so it might be nice to build more direct maps (here and narrower).  Well, for this wider code, the `$.map` is generating temp arrays for every comparison.  Perhaps `$.grep(wn[elr][src], (target) => target.t.includes(concept))` (or `wn[elr][src].some(...)` if we can support new enough JavaScript?) would avoid at least avoid the temp array.

Reply:
I should say: I don't feel strongly about this, just discussing.

Reply:
The reverse lookup was certainly very inefficient.  We already have `getChildRelationships` for the calculations code, so it makes sense to re-use that, and introduce an equivalent `getParentRelationships` which maintains a cache of parents indexed by child.  This is done in 3a8172453f244c01872d5d7cdb249599de8f248c.  I think the first call is O(anchorsInReport), and subsequent calls are O(anchorsForConcept), which seems reasonable.